src/main/java/io/confluent/connect/elasticsearch/ElasticsearchWriter.java

```diff
@@ -220,7 +239,22 @@ public void write(Collection<SinkRecord> records) {
       final IndexableRecord indexableRecord = converter.convertRecord(sinkRecord, index, type,
                                                                       ignoreKey, ignoreSchema);
 
-      bulkProcessor.add(indexableRecord, flushTimeoutMs);
+      // In the event that the sink record's value was null and the data converter has been told to IGNORE
+      // null values, the returned record will be null.
+      // TODO: If necessary, move the check for null-valued records to the top of the for-loop to avoid potential
+      //       performance penalties. Leaving the check here since it relegates all three kinds of null-record-handling
```

Comment:
I think "relegates" is more accurate but less common. Personally, I'd avoid this altogether with:

> Leaving the check here since DataConverter handles all of null-record behaviors

Reply:
AFAIU, `relegates` basically means `delegates to a lower level`, which is a useful clarification to make as for why delegating the handling of `null`-valued records to the `DataConverter` class is useful. The `ElasticsearchWriter` is high-level enough that it should be mostly concerned with just writing data, instead of deciding what to write (which is beneath it, and arguably more the responsibility of the `DataConverter`).

Reply:
According to my dictionary, `relegate` seems a bit too accurate, because it's interpretation as `downgrading` I don't think is particularly useful. I'd agree with @rhauch's suggestion to remove it altogether.